kafka-topics/exercise/plain/src/main/java/com/redhat/cdr/CDRProducer.java

```diff
@@ -0,0 +1,88 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+
+
+public class CDRProducer {
+
+    private static final Logger logger = LoggerFactory.getLogger(CDRProducer.class);
+
+    public static Properties getKafkaProperties() {
+
+        ClassroomConfig classroomConfig = getClassroomConfig();
+
+        Properties props = new Properties();
+
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, classroomConfig.getBoostrapServer() + ":"
+                + classroomConfig.getBootstrapPort());
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class.getName());
+        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+
+        props.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SSL");
+        props.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, classroomConfig.getWorkspacePath() + "/truststore.jks");
+        props.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, "password");
+
+        return props;
+    }
+
+    public static void main(String[] args) throws ExecutionException, InterruptedException {
+        Producer<Integer, String> producer = new KafkaProducer<>(getKafkaProperties());
+
+        for (int i = 1; i <= 7; i++) {
+
+            CallDetailRecord callDetailRecord = new CallDetailRecord(i, "Call record-" + i);
+
+            ProducerRecord<Integer, String> record = new ProducerRecord<>(
+                    "call-detail-records", callDetailRecord.getUserId(), callDetailRecord.toString()
+            );
+
+            producer.send(record, new Callback() {
+                public void onCompletion(RecordMetadata recordMetadata, Exception e) {
+                    // executes every time a record is successfully sent or an exception is thrown
+                    if (e == null) {
+                        // the record was successfully sent
+                        System.out.println("Sent record:");
+                        System.out.println("\tTopic = " + recordMetadata.topic());
+                        System.out.println("\tPartition = " + recordMetadata.partition());
+                        System.out.println("\tKey = " + record.key());
+                        System.out.println("\tValue = " + record.value());
+                    } else {
+                        logger.error("Error while producing", e);
+                    }
+                }
+            }).get();
+
+        }
+
+        producer.flush();
+        producer.close();
+    }
+
+    private static ClassroomConfig getClassroomConfig() {
+        ObjectMapper objectMapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        try {
+            return objectMapper.readValue(
+                    new File(System.getProperty("user.home") + "/.grading/ad482-workspace.json"),
```

Comment:
Is this path going to work on windows?

Reply:
I always forget about windows:) Let me check that in a vm. It should do practically

Reply:
It seems that it works:

`[main] ERROR com.redhat.cdr.CDRProducer - Make sure to run 'lab start eda-setup' in your workspace directory
java.io.FileNotFoundException: C:\Users\IEUser\.grading\ad482-workspace.json (Sistem belirtilen yolu bulamiyor)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
        at com.fasterxml.jackson.core.JsonFactory.createParser(JsonFactory.java:1029)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3413)
        at com.redhat.cdr.CDRProducer.getClassroomConfig(CDRProducer.java:79)
        at com.redhat.cdr.CDRProducer.getKafkaProperties(CDRProducer.java:27)
        at com.redhat.cdr.CDRProducer.main(CDRProducer.java:44)
Exception in thread "main" java.lang.NullPointerException
        at com.redhat.cdr.CDRProducer.getKafkaProperties(CDRProducer.java:31)
        at com.redhat.cdr.CDRProducer.main(CDRProducer.java:44)
`

Reply:
I will try this by installing the dynolabs and creating the relevant path. will let you know

Reply:
@jramcast Works like a charm. (I skippe the truststore generation part and it could able to access the directory and the kafka cluster,but gave an error for truststore since I did not create it)

Reply:
Great! Let's merge this then